rest/src/main/java/at/jku/isse/ecco/rest/RepositoryService.java

```diff
@@ -0,0 +1,264 @@
+package at.jku.isse.ecco.rest;
+
+import at.jku.isse.ecco.rest.classes.RestRepository;
+import at.jku.isse.ecco.service.EccoService;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.exceptions.HttpStatusException;
+import io.micronaut.http.multipart.CompletedFileUpload;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static at.jku.isse.ecco.rest.Settings.STORAGE_LOCATION_OF_REPOSITORIES;
+
+/** "Main" class called by all controllers.
+ * Holds all RepositoryHandler
+ * only one instance possible (singleton)
+ */
+
+public class RepositoryService {
+    private final Path repoStorage = Path.of(STORAGE_LOCATION_OF_REPOSITORIES);
+    private final Map<Integer, RepositoryHandler> repositories = new TreeMap<>();
+    private final AtomicInteger rId = new AtomicInteger();
+    private final EccoService generalService = new EccoService();
+    private static final RepositoryService instance;
+    private static final Logger LOGGER = Logger.getLogger(RepositoryService.class.getName());
+
+    // singleton
+    private RepositoryService() {
+        getRepositories();
+    }
+
+    static {
+        instance = new RepositoryService();
+        LOGGER.info("Repository service stated");
+    }
+
+    public static RepositoryService getInstance() {
+        return instance;
+    }
+
+    // Repositories ----------------------------------------------------------------------------------------------------
```

Comment:
same as above. If header comments are needed, the class does too much. Please refactor